diff --git a/node_modules/@notifee/react-native/android/src/main/java/io/invertase/notifee/NotifeeApiModule.java b/node_modules/@notifee/react-native/android/src/main/java/io/invertase/notifee/NotifeeApiModule.java
index 40bea64..ec4aa6f 100644
--- a/node_modules/@notifee/react-native/android/src/main/java/io/invertase/notifee/NotifeeApiModule.java
+++ b/node_modules/@notifee/react-native/android/src/main/java/io/invertase/notifee/NotifeeApiModule.java
@@ -36,16 +36,7 @@ public class NotifeeApiModule extends ReactContextBaseJavaModule implements Perm
     return Notifee.getInstance().getMainComponent(defaultComponent);
   }
 
-  // This method was removed upstream in react-native 0.74+, replaced with invalidate
-  // We will leave this stub here for older react-native versions compatibility
-  // ...but it will just delegate to the new invalidate method
-  public void onCatalystInstanceDestroy() {
-    invalidate();
-  }
-
-  // This method was added in react-native 0.74 as a replacement for onCatalystInstanceDestroy
-  // It should be marked @Override but that would cause problems in apps using older react-native
-  // When minimum supported version is 0.74+ add @Override & remove onCatalystInstanceDestroy
+  @Override
   public void invalidate() {
     NotifeeReactUtils.headlessTaskManager.stopAllTasks();
   }
